### ISSUE TITLE
sql: panic in distSQLPlanner.partitionSpans

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#16127]
https://sentry.io/cockroach-labs/cockroachdb/issues/280297374/

```
*errors.errorString: *errors.errorString v3.go:353
  File "github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go", line 353, in func1
  File "github.com/cockroachdb/cockroach/pkg/sql/executor.go", line 499, in func1
  File "github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go", line 433, in partitionSpans
  File "github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go", line 580, in createTableReaders
  File "github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go", line 1436, in createPlanForNode
...
(24 additional frame(s) were not displayed)

*errors.errorString v3.go:353
```

Assigning to @andreimatei for triage.

**Comment by RaduBerinde** [comment]
#16229 happens in a different context (CreateBackfiller) but it's the same crash.

**Comment by arjunravinarayan** [comment]
I did some initial investigation due to a second report in a second sentry issue.

The gist of it is that in `distSQLPlanner.partitionSpans()`, we seek through our spans looking to choose the right replica for each span. The following call happens:

```
replInfo, err := it.ReplicaInfo(ctx)
if err != nil {
	return nil, err
}
...
nodeID := replInfo.NodeDesc.NodeID
```
Which panics, presumably on a nil pointer exception because `NodeDesc` is not set. This means that the `ReplicaInfo` call is returning a malformed result (no err, but no valid NodeDesc). Chasing up the stack tree doesn't reveal anything immediately suspicious (there are a few interface calls with multiple implementations along the way), but this definitely should be addressed in 1.1.

**Comment by arjunravinarayan** [comment]
https://sentry.io/cockroach-labs/cockroachdb/issues/333831273/
